### DISCUSSION TITLE
Failed to connect to localhost/0:0:0:0:0:0:0:1:4317

### DISCUSSION THREAD

**Discussion by swapnil-kotwal-sp** [open-telemetry/opentelemetry-java-instrumentation#8100] [Q&A]
Hey Team,
I'm running my local Otel-Collector container on port 4343 as below
```
ed5c9ed0e8fd   otel/opentelemetry-collector-contrib:latest   "/otelcol-contrib --…"   8 minutes ago   
Up 8 minutes    0.0.0.0:1888->1888/tcp, 0.0.0.0:8888-8889->8888-8889/tcp, 0.0.0.0:13133->13133/tcp, 
55678-55679/tcp, 0.0.0.0:4343->4317/tcp   cloud-gateway-otel-collector-1
```

I've set Environment variable as `OTEL_EXPORTER_OTLP_ENDPOINT=http://otel_collector:4343/` where `otel_collector` is pointed to `127.0.0.1`
But, still  OpenTelemetry SDK not getting the endpoint, what am I missing here?

```
Mar 21, 2023 1:58:11 PM io.opentelemetry.sdk.internal.ThrottlingLogger doLog
SEVERE: Failed to export metrics. The request could not be executed. Full error message: Failed to connect to localhost/0:0:0:0:0:0:0:1:4317 
```

**Comment by laurit** [comment]
From the log message it looks like you are not using the java agent, is that correct? If so what are you using, spring starter?

**Reply by swapnil-kotwal-sp** [reply]
It is Java OT SDK only
```
private OpenTelemetrySdk initOpenTelemetry() {

        Resource resource =
                Resource.getDefault()
                        .merge(Resource.builder().put(SERVICE_NAME, "My Service").build());

        OpenTelemetrySdk openTelemetry =
                OpenTelemetrySdk.builder()
                        .setMeterProvider(
                                SdkMeterProvider.builder()
                                        .setResource(resource)
                                        .registerMetricReader(
                                                PeriodicMetricReader.builder(OtlpGrpcMetricExporter.getDefault())
                                                        .setInterval(Duration.ofMillis(1000))
                                                        .build())
                                        .build())
                        .buildAndRegisterGlobal();

        Runtime.getRuntime()
                .addShutdownHook(new Thread(((OpenTelemetrySdk) openTelemetry).getSdkMeterProvider()::shutdown));

        return openTelemetry;
    }
```


IT simply ignore every single flag.. I even have an ability to inject this vars to program directly as `JAVA_OPTS` but it doesn't cares those as well.

I was expecting when I specify `-Dotel.sdk.disabled=true` it should have stop all complaints of  `Failed to connect to localhost/0:0:0:0:0:0:0:1:4317` but no luck.  So, in my case it simply ignoring all `ENV_VARS` or `JAVA_OPTS` ?
```
JAVA_OPTS="$JAVA_OPTS -Xms256m -Xmx896m -Dsun.lang.ClassLoader.allowArraySyntax=true 
    -Dotel.metrics.exporter=otlp \
    -Dotel.exporter.otlp.endpoint=http://otel_collector:4343\
    -Dotel.exporter.otlp.protocol=grpc \
    -Dotel.service.name="my-service" \
    -Dotel.traces.exporter=none \
    -Dotel.sdk.disabled=true
"
java $JAVA_OPTS \
      com.mypackage.Main
}
```

I'm using below SDK versions.
```
implementation "io.opentelemetry:opentelemetry-sdk:1.23.1"
implementation "io.opentelemetry:opentelemetry-exporter-otlp:1.23.1"
implementation "io.opentelemetry:opentelemetry-semconv:1.23.1-alpha"
```

**Reply by swapnil-kotwal-sp** [reply]
do I need to specifically `setEndPoint("http://otel_collector:4343");` here? Or shall I go for `Auto Configuration` ? 🤔 https://opentelemetry.io/docs/instrumentation/java/manual/#auto-configuration

**Reply by laurit** [reply]
These flags belong to the `AutoConfiguredOpenTelemetrySdk`, see https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md. If you are using `OpenTelemetrySdk` none of these will work.

**Reply by swapnil-kotwal-sp** [reply]
Yeah, just realised it. Just curious more, how endpoints are set in `Manual Instrumentation`?

**Reply by laurit** [reply]
You could set it on the exporter, for example see https://github.com/open-telemetry/opentelemetry-java/blob/38bf4d8ec0d794f9be77d2fe33b1ed99f5262eef/integration-tests/otlp/src/main/java/io/opentelemetry/integrationtest/OtlpExporterIntegrationTest.java#L185
